kratos/includes/fill_communicator.h

```diff
@@ -0,0 +1,251 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_FILL_COMMUNICATOR_H_INCLUDED )
+#define  KRATOS_FILL_COMMUNICATOR_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+
+///@}
+///@name  Functions
+///@{
+
+
+///@}
+///@name Kratos Classes
+///@{
+
+// Forward declaration of ModelPart
+class ModelPart;
```

Comment:
I think we said that we should not use forward declarations
Can you change it still?

Reply:
mmm... this was in the original `ParallelFillCommunicator`. I assumed we intentionally did it in there to avoid including the `ModelPart`.

Reply:
Yes this was done, but then we decided not to use forward declarations any more as they can be problematic

I know it is still used in some cases, but eventually we should remove those

Outlook: with C++20 modules afaik this becomes irrelevant anyway as the include/import mechanism changes

Reply:
OK. I'll remove it then.